src/actions/AddPeriodicRelativeNodalDifference.C

```diff
@@ -0,0 +1,434 @@
+                              var_log_name + "_previous_cycle_NodalAverage",
+                              true);
+    }
+
+    for (unsigned int cur_num = 0; cur_num < number_var; cur_num++)
+    {
+      std::string var_name = var[cur_num];
+
+      addPerviousSolutionsKernels(var_name + "_previous_cycle_sudo", var_name);
+      addPerviousSolutionsKernels(var_name + "_previous_cycle", var_name + "_previous_cycle_sudo");
+
+      addNormalizationKernels(
+          var_name + "_normalized", var_name, var_name + "_NodalAverage", false);
+      addNormalizationKernels(var_name + "_previous_cycle_normalized",
+                              var_name + "_previous_cycle",
+                              var_name + "_previous_cycle_NodalAverage",
+                              false);
+    }
+  }
+
+  // Adding Postprocessor to calculate nodal average and difference
+  else if (_current_task == "add_postprocessor")
+  {
+    for (unsigned int cur_num = 0; cur_num < number_var_log; cur_num++)
+    {
+      std::string var_log_name = var_log[cur_num];
+
+      addAverageNodalPP(var_log_name, true);
+      addAverageNodalPP(var_log_name + "_previous_cycle", true);
+
+      addRelativePeriodicDiffPP(
+          var_log_name + "_normalized", var_log_name + "_previous_cycle_normalized", var_log_name);
+    }
+
+    for (unsigned int cur_num = 0; cur_num < number_var; cur_num++)
+    {
+      std::string var_name = var[cur_num];
+
+      addAverageNodalPP(var_name, false);
+      addAverageNodalPP(var_name + "_previous_cycle", false);
+
+      addRelativePeriodicDiffPP(
+          var_name + "_normalized", var_name + "_previous_cycle_normalized", var_name);
+    }
+  }
+
+  // Adding a periodic control, so Auxkernels only are on once par cycle
+  else if (_current_task == "add_control")
+  {
+    for (unsigned int i = 0; i < _num_controller_set; ++i)
+    {
+      for (MooseIndex(_enable_start) j = 0; j < _enable_start.size(); ++j)
+      {
+        _enable_start_start_time_index[j] = _start_time + _period * i;
+        _enable_start_end_time_index[j] = _start_time + _period * i + (_period * 0.0001);
+      }
+      for (MooseIndex(_enable_end) j = 0; j < _enable_end.size(); ++j)
+      {
+        _enable_end_start_time_index[j] =
+            (_start_time + _period) + _period * i + (_period * 0.0001);
+        _enable_end_end_time_index[j] = (_start_time + _period) + _period * i + (_period * 0.0002);
+      }
+
+      if (_enable_start.size() > 0)
+      {
+        if (i == 0)
+        {
+          InputParameters params = _factory.getValidParams("TimePeriod");
+          params.set<std::vector<std::string>>("enable_objects") = {_enable_start};
+          params.set<std::vector<Real>>("start_time") = _enable_start_start_time_index;
+          params.set<std::vector<Real>>("end_time") = _enable_start_end_time_index;
+          params.set<ExecFlagEnum>("execute_on", true) = {EXEC_INITIAL, EXEC_TIMESTEP_BEGIN};
```

Comment:
See my comment from `AddPeriodicControllers.C`. Same applies in the if statements in this file.